src/Serialization/Groups.jl

```diff
@@ -277,3 +277,30 @@ end
 function load_object(s::DeserializerState, ::Type{FinGenAbGroupElem}, G::FinGenAbGroup)
   return G(vec(load_object(s, Matrix{ZZRingElem})))
 end
+
+##############################################################################
+# MatrixGroup
+
+@register_serialization_type MatrixGroup uses_id
+
+type_params(G::MatrixGroup) = TypeParams(MatrixGroup,
+                                         :base_ring => base_ring(G),
+                                         :degree => degree(G))
+
+save_object(s::SerializerState, G::MatrixGroup) = save_object(s, matrix.(gens(G)))
+
+function load_object(s::DeserializerState, ::Type{MatrixGroup}, params::Dict)
+  R = params[:base_ring]
+  d = params[:degree]
+  generators = load_object(s, Vector{Matrix{elem_type(R)}}, R)
```

Comment:
```suggestion
  generators = load_object(s, Vector{dense_matrix_type(R)}, R)
```
more efficient code due to concrete elem type

Reply:
apparently FqMatrix is not a Matrix, so this doesn't do the same thing

Reply:
I think it makes more sense to store as Matrix as opposed to MatElem, since then we dont need to store the MatSpace

Reply:
ah, but then it's still better to load as MatElem though. ok there is an easy fix